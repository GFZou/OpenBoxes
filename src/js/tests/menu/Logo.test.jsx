import React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from 'store';

import Logo from 'components/Layout/Logo';

it('test if logo matches snapshot', () => {
  const renderedLogo = renderer.create(<Router><Logo logoUrl="" store={store} /></Router>).toJSON();
  expect(renderedLogo).toMatchSnapshot();
});

it('test if logo correctly rendering', () => {
  const renderedLogo = renderer.create(<Router><Logo logoUrl="" store={store} /></Router>).root;
  expect(renderedLogo.findByProps({ className: 'd-flex align-items-center logo-wrapper' }))
    .toBeTruthy();
});

it('test logo after redirecting to main page', () => {
  render(<Router><Logo logoUrl="" store={store} /></Router>);
  const image = screen.getByAltText('Openboxes');
  expect(screen.getByAltText('Openboxes')).toBeTruthy();
  fireEvent.click(image);
  expect(screen.getByAltText('Openboxes')).toBeTruthy();
});
