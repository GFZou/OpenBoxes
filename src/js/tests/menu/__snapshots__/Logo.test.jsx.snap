// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if logo matches snapshot 1`] = `
<div
  className="d-flex align-items-center logo-wrapper"
>
  <div
    className="logo-square"
  >
    <a
      href="/openboxes"
      onClick={[Function]}
    >
      <img
        alt="Openboxes"
        height="40"
        src=""
        width="40"
      />
    </a>
  </div>
</div>
`;

exports[`test redirecting to main page 1`] = `
<div
  className="d-flex align-items-center logo-wrapper"
>
  <div
    className="logo-square"
  >
    <a
      href="/openboxes"
      onClick={[Function]}
    >
      <img
        alt="Openboxes"
        height="40"
        src=""
        width="40"
      />
    </a>
  </div>
</div>
`;

exports[`test redirecting to main page 2`] = `
<img
  alt="Openboxes"
  height="40"
  src=""
  width="40"
/>
`;
